File: Ranglistenpunkte/RankingDetailViewController.m
//
//  RankingDetailViewController.m
//  Ranglistenpunkte
//
//  Created by Martin Kautz on 24.11.14.
//  Copyright (c) 2014 Martin Kautz. All rights reserved.
//

#import "RankingDetailViewController.h"
#import "RegattaDetailCell.h"
#import "SailorCell.h"

@interface RankingDetailViewController() <UITableViewDataSource, UITableViewDelegate>
// ---------------------------------------------------------------------------------------------------------------------
@property (nonatomic, weak)     IBOutlet    UITableView                 *theTableView;
// ---------------------------------------------------------------------------------------------------------------------
@end

@implementation RankingDetailViewController


//----------------------------------------------------------------------------------------------------------------------
#pragma mark - Init & lifecycle
//----------------------------------------------------------------------------------------------------------------------
- (void)viewDidLoad
{
    [super viewDidLoad];
    self.theTableView.delegate      = self;
    self.theTableView.dataSource    = self;
    self.title                      = @"Detail";
}


//----------------------------------------------------------------------------------------------------------------------
#pragma mark - UITableViewDataSorce protocol methods
//----------------------------------------------------------------------------------------------------------------------
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    switch (section) {
        case 0:
            return 1;
            break;

        default:
        {
            NSArray *aRegattas = self.theDataSource[@"regatta"];
            if (aRegattas == nil) {
                return 1;
            } else {
                int counter = 0;
                NSDictionary *regatta;
                for (regatta in aRegattas) {
                    if (regatta.count > 0) {
                        counter ++;
                    }
                }
                return counter;
            }
        }
            break;
    }
}

// ---------------------------------------------------------------------------------------------------------------------
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 2;
}

// ---------------------------------------------------------------------------------------------------------------------
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    switch (indexPath.section) {
        case 0:
        {
            SailorCell *cell                = (SailorCell *)[self.theTableView dequeueReusableCellWithIdentifier:@"SailorCell"
                                                                                                    forIndexPath:indexPath];
            NSDictionary *sailor            = self.theDataSource;
            cell.firstnamenameLabel.text    = [NSString stringWithFormat:@"%@ %@", sailor[@"firstname"], sailor[@"name"]];
            cell.posLabel.text              = [NSString stringWithFormat:@"%@", sailor[@"pos"]];
            cell.clubLabel.text             = [NSString stringWithFormat:@"%@", sailor[@"club"]];
            cell.sailCountryLabel.text      = [NSString stringWithFormat:@"%@", sailor[@"sailCountry"]];
            cell.sailNumberLabel.text       = [NSString stringWithFormat:@"%@", sailor[@"sailNumber"]];
            cell.totalPointsLabel.text      = [NSString stringWithFormat:@"%.2f", [sailor[@"totalPoints"]floatValue]];

            cell.yobLabel.text              = [NSString stringWithFormat:@"%@", sailor[@"yob"]];
            cell.totalRunsLabel.text        = [NSString stringWithFormat:@"%@", sailor[@"totalRuns"]];
            cell.selectionStyle             = UITableViewCellSelectionStyleNone;

            return cell;
        }
            break;

        default:
        {
            RegattaDetailCell *cell         = (RegattaDetailCell *)[self.theTableView dequeueReusableCellWithIdentifier:@"RegattaDetailCell"
                                                                                                           forIndexPath:indexPath];
            NSDictionary *regatta           = self.theDataSource[@"regatta"][indexPath.row];
            cell.rnamelabel.text            = [NSString stringWithFormat:@"%@", regatta[@"rname"]];
            cell.sl_pointsLabel.text        = [NSString stringWithFormat:@"%.2f", [regatta[@"sl_points"]floatValue]];
            cell.sl_points_cupLabel.text    = [NSString stringWithFormat:@"%.2f", [regatta[@"sl_points_cup"]floatValue]];
            cell.positionboatsLabel.text    = [NSString stringWithFormat:@"%@/%@", regatta[@"position"], regatta[@"boats"]];
            cell.runs_totalLabel.text       = [NSString stringWithFormat:@"%@", regatta[@"runs_total"]];
            cell.runs_scoredLabel.text      = [NSString stringWithFormat:@"%@", regatta[@"runs_scored"]];
            cell.selectionStyle             = UITableViewCellSelectionStyleNone;

            return cell;
        }
            break;
    }
    return nil;

}

//----------------------------------------------------------------------------------------------------------------------
#pragma mark - UITableViewDelegate protocol methods
//----------------------------------------------------------------------------------------------------------------------
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
{
    return indexPath.section == 0 ? 120.0f : 80.0f;
}

@end

File: Ranglistenpunkte/RankingDetailViewController.swift
//
//  RankingDetailViewController.swift
//  Ranglistenpunkte
//
//  Created by Martin Kautz on 26.11.14.
//  Copyright (c) 2014 Martin Kautz. All rights reserved.
//

import UIKit

class RankingDetailViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {

    @IBOutlet weak var theTableView:UITableView!
    var theDataSource = NSDictionary()

    // -----------------------------------------------------------------------------------------------------------------
    // MARK: - Init & lifecycle
    // -----------------------------------------------------------------------------------------------------------------
    override func viewDidLoad() {
        super.viewDidLoad()
        theTableView.delegate   = self
        theTableView.dataSource = self;
        title                   = "Detail"
    }


    // -----------------------------------------------------------------------------------------------------------------
    // MARK: - UITableViewDataSource protocol methods
    // -----------------------------------------------------------------------------------------------------------------
    func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 2
    }

    // -----------------------------------------------------------------------------------------------------------------
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        switch section {
        case 0:
            return 1
        default:
            if let regattas = theDataSource["regatta"] as? NSArray {
                var counter:Int = 0
                for elem in regattas {
                    if let regatta = elem as? NSDictionary {
                        if let regattaId = regatta["regattaId"] as? String {
                            counter++
                        }
                    }
                }
                return counter
            } else {
                return 1
            }
        }
    }

    // -----------------------------------------------------------------------------------------------------------------
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {

        switch indexPath.section {

        case 0:
            var cell:SailorCell = SailorCell()
            cell = tableView.dequeueReusableCellWithIdentifier("SailorCell", forIndexPath: indexPath) as SailorCell
            cell.selectionStyle = .None
            // ---------------------------------------------------------------------------------------------------------
            cell.firstnamenameLabel.text    = (theDataSource["firstname"] as String) + " " + (theDataSource["name"] as String)
            cell.clubLabel.text             = theDataSource["club"] as? String
            cell.sailCountryLabel.text      = theDataSource["sailCountry"] as? String
            cell.sailNumberLabel.text       = theDataSource["sailNumber"] as? String
            // ---------------------------------------------------------------------------------------------------------
            var yob: String
            if let y = theDataSource["yob"] as? String {
                yob = y
            } else if let y = theDataSource["yob"] as? Int {
                yob = String(y)
            } else {
                yob = "-"
            }
            cell.yobLabel.text = yob
            // ---------------------------------------------------------------------------------------------------------
            var totalPoints: String
            if let tp = theDataSource["totalPoints"] as? String {
                totalPoints = tp
            } else if let tp = theDataSource["totalPoints"] as? Float {
                totalPoints = NSString(format:"%.2f", tp)
            } else {
                totalPoints = "-"
            }
            cell.totalPointsLabel.text = totalPoints
            // ---------------------------------------------------------------------------------------------------------
            var totalRuns : String
            if let tr = theDataSource["totalRuns"] as? String {
                totalRuns = tr
            } else if let tr = theDataSource["totalRuns"] as? Int {
                totalRuns = String(tr)
            } else {
                totalRuns = "-"
            }
            cell.totalRunsLabel.text = totalRuns
            // ---------------------------------------------------------------------------------------------------------
            var pos: String
            if let p = theDataSource["pos"] as? String {
                pos = p
            } else if let p = theDataSource["pos"] as? Int {
                pos = String(p)
            } else {
                pos = "-"
            }
            cell.posLabel.text = pos
            // ---------------------------------------------------------------------------------------------------------
            return cell

        default:
            let regatta = theDataSource["regatta"]?[indexPath.row] as NSDictionary
            var cell:RegattaDetailCell = RegattaDetailCell()
            cell = tableView.dequeueReusableCellWithIdentifier("RegattaDetailCell", forIndexPath: indexPath) as RegattaDetailCell
            cell.selectionStyle = .None
            // ---------------------------------------------------------------------------------------------------------
            cell.rnamelabel.text            = regatta["rname"] as? String
            cell.sl_pointsLabel.text        = regatta["sl_points"] as? String
            cell.sl_points_cupLabel.text    = regatta["sl_points_cup"] as? String
            // ---------------------------------------------------------------------------------------------------------
            var position : String
            if let pos = regatta["position"] as? String {
                position = pos
            } else if let pos = regatta["position"] as? Int {
                position = String(pos)
            } else {
                position = "-"
            }
            // ---------------------------------------------------------------------------------------------------------
            var boats : String
            if let b = regatta["boats"] as? String {
                boats = b
            } else if let b = regatta["boats"] as? Int {
                boats = String(b)
            } else {
                boats = "-"
            }
            cell.positionboatsLabel.text    = position + "/" + boats
            // ---------------------------------------------------------------------------------------------------------
            var total: String
            if let t: String = regatta["runs_total"] as? String {
                total = t
            } else if let t: Int = regatta["runs_total"] as? Int {
                total = String(t)
            } else {
                total = "-"
            }
            cell.runs_totalLabel.text = total
            // ---------------------------------------------------------------------------------------------------------
            var scored: String
            if let s: String = regatta["runs_scored"] as? String {
                scored = s
            } else if let s: Int = regatta["runs_scored"] as? Int {
                scored = String(s)
            } else {
                scored = "-"
            }
            cell.runs_scoredLabel.text = scored
            // ---------------------------------------------------------------------------------------------------------
            return cell
        }
    }

    // -----------------------------------------------------------------------------------------------------------------
    // MARK: - UITableViewDelegate protocol methods
    // -----------------------------------------------------------------------------------------------------------------
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return indexPath.section == 0 ? 120.0 : 80.0
    }

}
